include/NovelRT.Interop/Windowing/NrtIWindowingDevice.h

```diff
@@ -0,0 +1,25 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+#ifndef NOVELRT_INTEROP_WINDOWING_NRTIWINDOWINGDEVICE_H
+#define NOVELRT_INTEROP_WINDOWING_NRTIWINDOWINGDEVICE_H
+
+#include "../NrtTypedefs.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    NrtResult Nrt_IWindowingDevice_Destroy(NrtIWindowingDeviceHandle device);
+    NrtBool Nrt_IWindowingDevice_GetIsVisible(NrtIWindowingDeviceHandle device);
+    NrtBool Nrt_IWindowingDevice_GetShouldClose(NrtIWindowingDeviceHandle device);
+    // The returned string is allocated, remember to free it!
+    char* Nrt_IWindowingDevice_GetWindowTitle(NrtIWindowingDeviceHandle device);
```

Comment:
It's quite hard currently to have consistent `char*` string allocation conventions, since `IWindowingDevice::GetWindowTitle` returns an `std::string` which inevitably copies the string, but other APIs don't return a string by copy and instead return it by reference (such as `Nrt_getLastError`).

Reply:
Possible bug in C++ then?

Reply:
Maybe? In places that we have a `std::string`, i would thing we should be returning using `.c_str()` which gives us a `const char*`...

It's also odd because we don't use it much, but the entirety of the error handling API calls upon pre-defined `const char*`s

Reply:
Also another place to point out - direct output here? Or should we shift to a `NrtResult` return with an output parameter?